### PR TITLE
split RemoteAssetGraph in to RemoteRepository and RemoteGlobal

### PR DESCRIPTION
**Pull request by alangenfeld**
More accurately model the two flavors of `RemoteAssetGraph` that we create, a repository scoped one and a global / workspace scoped one. The global asset graph is built by combining all of the repository scoped asset graphs.

## How I Tested These Changes

existing coverage

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#25199** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25199?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by smackesey — COMMENTED** [review]
Thoughts on `RemoteWorkspaceAssetGraph` (`Node`) instead of `RemoteGlobalAssetGraph` (`Node`)? Feels like it better matches `RemoteRepositoryAssetGraph`.

**Comment by alangenfeld** [comment]
> Thoughts on RemoteWorkspaceAssetGraph (Node) instead of RemoteGlobalAssetGraph (Node)? Feels like it better matches RemoteRepositoryAssetGraph.

yea i went back and forth on it - agree workspace matches the code artifacts better. Global is the term that gets used in the product / conversationally. I would be good with workspace.

**Comment by smackesey** [comment]
>yea i went back and forth on it - agree workspace matches the code artifacts better. Global is the term that gets used in the product / conversationally. I would be good with workspace.

I think if it were standalone "global" would be better but since this is contrasted with "repository" I have a strong pref for "workspace". Stamping, leave final decision to you

**Review by smackesey — APPROVED** [review]

**Comment by alangenfeld** [comment]
I like your argument, will rename

File: python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py [hunk 1]
```diff
@@ -442,6 +430,118 @@ def get_implicit_job_name_for_assets(
...
+
+        for check_snap in repo.get_asset_check_node_snaps():
+            id = check_snap.execution_set_identifier
+            key = check_snap.key
+
+            asset_checks_by_key[key] = RemoteAssetCheckNode(
+                handle=repo.handle,
+                asset_check=check_snap,
+                execution_set_entity_keys=execution_sets_by_id[id] if id is not None else {key},
+            )
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py:502]: [B]

File: python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py [hunk 2]
```diff
@@ -451,73 +551,95 @@ def split_entity_keys_by_repository(
...
-        if id is not None:
-            execution_sets_by_id[id].add(key)
-
-    execution_sets_by_key: Dict[EntityKey, Set[EntityKey]] = {}
-    for item in all_items:
-        id = item.execution_set_identifier
-        key = item.asset_key if isinstance(item, AssetNodeSnap) else item.key
-        execution_sets_by_key[key] = execution_sets_by_id[id] if id is not None else {key}
-
-    return execution_sets_by_key
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py:523]: [A]

**Pull request merged**